tests/data/contracts.py

```diff
@@ -138,3 +138,508 @@
     return (arg,);
 }
 """
+CAIRO_BINDINGS_CONTRACT_ENUM = """\
+enum MyEnumShort { a: felt, b: felt }
+enum MyEnumLong { a: felt, b: felt, c: felt }
+enum MyEnumGeneric<S, T> { a: T, b: S, c: T }
+fn main() -> felt {
+    let es0 = MyEnumShort::a(10);
+    match_short(es0);
+    let es1 = MyEnumShort::b(11);
+    match_short(es1);
+
+    let el0 = MyEnumLong::a(20);
+    match_long(el0);
+    let el1 = MyEnumLong::b(21);
+    match_long(el1);
+    let el2 = MyEnumLong::c(22);
+    match_long(el2);
+    let eg1: MyEnumGeneric::<(), felt> = MyEnumGeneric::<(), felt>::a(30);
+    let eg2: MyEnumGeneric::<(), felt> = MyEnumGeneric::<(), felt>::b(());
+    let eg3: MyEnumGeneric::<(), felt> = MyEnumGeneric::<(), felt>::c(32);
+    300
+}
+
+fn match_short(e: MyEnumShort) -> felt {
+    match e {
+        MyEnumShort::a(x) => {
+            x
+        },
+        MyEnumShort::b(x) => {
+            x
+        },
+    }
+}
+
+fn match_long(e: MyEnumLong) -> felt {
+    match e {
+        MyEnumLong::a(x) => {
+            x
+        },
+        MyEnumLong::b(x) => {
+            x
+        },
+        MyEnumLong::c(x) => {
+            x
+        },
+    }
+}
+"""
+CAIRO_BINDINGS_CONTRACT_STARKNET_HELLO = """\
+#[contract]
+mod HelloStarknet {
+    struct Storage { balance: felt, }
+
+    // Increases the balance by the given amount.
+    #[external]
+    fn increase_balance(amount: felt) {
+        balance::write(balance::read() + amount);
+    }
+
+    // Returns the current balance.
+    #[view]
+    fn get_balance() -> felt {
+        balance::read()
+        0
+    }
+}
+"""
+CAIRO_BINDINGS_CONTRACT_STARKNET_HELLO_TEST = """\
+#[abi]
+trait IAnotherContract {
+fn foo(); }
+
+
+#[contract]
+mod TestContract {
+    struct Storage { my_storage_var: felt, }
+
+    fn internal_func() -> felt {
+        1
+    }
+
+    #[external]
+    fn test(ref arg: felt, arg1: felt, arg2: felt) -> felt {
+        let x = my_storage_var::read();
+        my_storage_var::write(x + 1);
+        x + internal_func()
+    }
+
+    #[external]
+    fn empty() {
+    }
+}
+"""
+CAIRO_BINDINGS_TESTS = """\
```

Comment:
You can use cairo1_test_suite_parser to run those tests (you can try using roll cheatcode inside). An example how to do it is `tests/integration/cairo1/parser_test.py`

Reply:
I think we should not do this here. It's good to have tests that only check if the compilation is OK. I can do this in a separate test that uses both the parser and the compiler.

Reply:
[add tests for compiler+parser](https://github.com/software-mansion/protostar/pull/1316/commits/aa9a39455f7c1641671744f4717804bf86379de7)

Reply:
I am okay with that, It was just a hint 😉